// Copyright (C) 2017 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "vector3.proto";
import "quaternion.proto";

message BoneBuffer {
    uint32               Parent         = 1;
    Vector3Buffer        Scale          = 2;
    QuaternionBuffer     Rotation       = 3;
    Vector3Buffer        Translation    = 4;
}

message FrameBuffer {
    repeated BoneBuffer Bone = 1;
}

message AnimationBuffer {
	uint32 Version                      = 1;
    uint32 FrameRate                    = 2;
    float Duration                      = 3;
    repeated FrameBuffer Frame          = 4;
}
